scripts/genpinctrl/config.yaml

```diff
@@ -0,0 +1,139 @@
+# Configuration file for pinctrl generation (except F1 series).
+#
+# This file contains a list of pin configuration templates used to generate the
+# pinctrl files. Each entry can have the following fields:
+#
+#   - name (mandatory): This is the pin alternate function name, e.g. UART_TX.
+#     It is used to group pin configurations alphabetically in the generated
+#     pinctrl files.
+#
+#   - match (mandatory): This is a regular expression used to match against
+#     CubeMX pin configuration names. The regular expression should be as
+#     precise as possible. Note that it needs to be escaped here in the
+#     configuration file.
+#
+#   - bias (optional): Bias setting (disable, pull-up, pull-down). If not set
+#     pinctrl file will not contain the bias entry, equivalent to disable
+#     setting.
+#
+#   - drive (optional): Drive setting (push-pull, open-drain). If not set
+#     pinctrl file will not contain the drive entry, equivalent to push-pull
+#     setting
+#
+#   - variant (optional): Define another variant of a pin configuration. This
+#     setting can be useful to define multiple configurations for a certain
+#     alternate function.
+#
+
+---
+- name: ADC_IN / ADC_INN / ADC_INP
+  match: "^ADC(?:\\d+)?_IN[NP]?\\d+$"
+
+- name: CAN_RX
+  match: "^CAN\\d+_RX$"
+  bias: pull-up
+
+- name: CAN_TX
+  match: "^CAN\\d+_TX$"
+
+- name: DAC_OUT
+  match: "^DAC(?:\\d+)?_OUT\\d+$"
+
+- name: FDCAN_RX
+  match: "^FDCAN\\d+_RX$"
+  bias: pull-up
+
+- name: FDCAN_TX
+  match: "^FDCAN\\d+_TX$"
+
+- name: I2C_SCL
+  match: "^I2C\\d+_SCL$"
+  drive: open-drain
+  bias: pull-up
+
+- name: I2C_SDA
+  match: "^I2C\\d+_SDA$"
+  drive: open-drain
+  bias: pull-up
+
+- name: I2S_CK
+  match: "^I2S\\d+_CK$"
+
+- name: I2S_WS
+  match: "^I2S\\d+_WS$"
+
+- name: I2S_SD
+  match: "^I2S\\d+_SD$"
+
+- name: SPI_MASTER_MISO
+  match: "^SPI\\d+_MISO$"
+  bias: pull-down
+  variant: master
+
+- name: SPI_MASTER_MOSI
+  match: "^SPI\\d+_MOSI$"
+  bias: pull-down
+  variant: master
+
+- name: SPI_MASTER_SCK
+  match: "^SPI\\d+_SCK$"
+  bias: pull-down
+  variant: master
+
+- name: SPI_MASTER_NSS
+  match: "^SPI\\d+_NSS$"
+  bias: pull-up
+  variant: master
+
+- name: SPI_SLAVE_MISO
+  match: "^SPI\\d+_MISO$"
+  bias: pull-down
+  variant: slave
+
+- name: SPI_SLAVE_MOSI
+  match: "^SPI\\d+_MOSI$"
+  bias: pull-down
+  variant: slave
+
+- name: SPI_SLAVE_SCK
+  match: "^SPI\\d+_SCK$"
+  bias: pull-down
+  variant: slave
+
+- name: SPI_SLAVE_NSS
+  match: "^SPI\\d+_NSS$"
+  bias: pull-up
+  variant: slave
```

Comment:
Is it worth adding slave/master config since there is no diff?

Reply:
F1 series need different configs for master/slave, as MISO/MOSI will be either alternate/input depending on operation mode. Other series don't need that, what should we do?

Reply:
This is one of the interest of separating the config files. I think they should do whatever makes sense:
If no variant is needed, let's not add it.

Reply:
I have added `SPI_MISO/MOSI/SCK/NSS` for non-F1 and master/slave variants for F1, let me know if it's ok now.

Reply:
This is fine, txs